modules/hostap/src/supp_api.h

```diff
@@ -138,6 +138,17 @@ int supplicant_filter(const struct device *dev, struct wifi_filter_info *filter)
 int supplicant_channel(const struct device *dev, struct wifi_channel_info *channel);
 
 #ifdef CONFIG_AP
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+/**
+ * @brief Get Wi-Fi AP Status
+ *
+ * @param dev Wi-Fi device
+ * @param params AP status
+ * @return 0 for OK; -1 for ERROR
+ */
+int hapd_state(const struct device *dev, int *state);
+#endif
```

Comment:
I suggest to provide the same public API whatever the value of `CONFIG_WIFI_NM_HOSTAPD_AP`:
```
#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
int hapd_state(const struct device *dev, int *state);
#else
static inline int hapd_state(const struct device *dev, int *state) {
    return -EINVAL;
}
#endif
```

Thus the caller does not have to add `#ifdef` around the call to this function.

Reply:
Updated.